MapleBacon/Core/MapleBacon.swift

```diff
@@ -42,9 +42,11 @@ public final class MapleBacon {
                           progress: DownloadProgress?,
                           completion: ImageDownloadCompletion?) {
     let key = url.absoluteString
-    cache.retrieveImage(forKey: key, transformerId: transformer?.identifier) { image, _ in
+    cache.retrieveImage(forKey: key, transformerId: transformer?.identifier) { [weak self] image, _ in
```

Comment:
For the regular use case this isn't strictly needed because MapleBacon is a singleton that lives forever anyway, as such there's no issue if this strongly retains.
However, could you do me the favour of merging the guard statements into one, that is
```swift
guard let self = self, let image = image else {
```